src/layman/layer/geoserver/sld.py

```diff
@@ -79,18 +80,23 @@ def ensure_custom_sld_file_if_needed(workspace, layer):
     input_mask_flags = file_dict['mask_flags']
     norm_file_dict = file_dict['normalized_file']
     norm_stats = norm_file_dict['stats']
+    norm_nodata_value = norm_file_dict['nodata_value']
 
     # if there is one grayscale band without mask flags
-    if input_color_interpretations == ['Gray'] and input_mask_flags == [{gdalconst.GMF_ALL_VALID}]:
+    if input_color_interpretations == ['Gray'] and input_mask_flags[0] <= {gdalconst.GMF_ALL_VALID, gdalconst.GMF_NODATA}:
```

Comment:
I think  `input_mask_flags[0].issubset({gdalconst.GMF_ALL_VALID, gdalconst.GMF_NODATA})` is readable.